core/modules/redirect/redirect.install

```diff
@@ -172,3 +175,15 @@ function redirect_update_1003() {
 
   db_query("UPDATE {redirect} SET status_code = :default_status WHERE status_code = 0", array(':default_status' => $default_status));
 }
+
+/**
+ * Fix redirect table schema of upgraded sites.
+ */
+function redirect_update_1004() {
+  if (db_field_exists('redirect', 'status')) {
+    if (db_index_exists('redirect', 'status_source_language')) {
```

Comment:
could there be any edge cases where this column exists without the `status` column? Shouldn't we be checking for both of these and deleting them independently?

Reply:
That's actually an index, not a column. It might exist or not, but not if the column's not there.

Both, the column and the index won't exist on a native Backdrop install. When upgrading from Dupal, we should be cautious and not assume that the schema is exactly as in D7 hook_schema. It might have changed over time.